### PR TITLE
Checkbox component

### PR DESCRIPTION
**Pull request by liondedan**
**Description**
- Creation of a checkbox component
- Associated tests

**Links**
https://sofarsounds.atlassian.net/browse/TBC-220

### CODE REVIEW

**Comment by davidfloegel** [comment]
# GitHub Pages Deployment
Pull Request:
https://sofarsounds.github.io/maestro/pull/168
Branch:
https://sofarsounds.github.io/maestro/branches/checkbox-component
Storybook:
https://sofarsounds.github.io/maestro/05c77f77a981f0a93b41ec7887f384f7807fe900/
Build:
https://sofarsounds.github.io/maestro/05c77f77a981f0a93b41ec7887f384f7807fe900/dist.zip

**Comment by davidfloegel** [comment]
# GitHub Pages Deployment
Pull Request:
https://sofarsounds.github.io/maestro/pull/168
Branch:
https://sofarsounds.github.io/maestro/branches/checkbox-component
Storybook:
https://sofarsounds.github.io/maestro/d3d7a05502c4dfeeffa7cf945d94dda2134a555b/
Build:
https://sofarsounds.github.io/maestro/d3d7a05502c4dfeeffa7cf945d94dda2134a555b/dist.zip

**Comment by davidfloegel** [comment]
# GitHub Pages Deployment
Pull Request:
https://sofarsounds.github.io/maestro/pull/168
Branch:
https://sofarsounds.github.io/maestro/branches/checkbox-component
Storybook:
https://sofarsounds.github.io/maestro/16497c3a8e1fd3591f3147c14d082d0d398bbbc7/
Build:
https://sofarsounds.github.io/maestro/16497c3a8e1fd3591f3147c14d082d0d398bbbc7/dist.zip

**Comment by davidfloegel** [comment]
# GitHub Pages Deployment
Pull Request:
https://sofarsounds.github.io/maestro/pull/168
Branch:
https://sofarsounds.github.io/maestro/branches/checkbox-component
Storybook:
https://sofarsounds.github.io/maestro/89463264d9e9e812d89b1e0b2dc295e082716597/
Build:
https://sofarsounds.github.io/maestro/89463264d9e9e812d89b1e0b2dc295e082716597/dist.zip

File: src/atoms/Checkbox/Checkbox.test.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+  it.skip('cannot be toggled when disabled', () => {
+    // A few options tried below - enzyme does not actually update the DOM
+    // const wrapper = setup({ checked: false, disable: true }, 'Check MY Box');
+    // const input = wrapper.find('input');
+    // input.simulate('click');
+    // wrapper.find('input').simulate('change', { target: { checked: true } });
+    // wrapper.getDOMNode().dispatchEvent(new Event('click'));
+    // wrapper.find('input').simulate('change', { target: { checked: true } });
+    // expect(mockClick).toHaveBeenCalled();
```
Comment (adaam2) [inline src/atoms/Checkbox/Checkbox.test.tsx]: This is a weird ass test failure. I just looked into it and wtf mate

**Pull request merged**